templates/messaging/user_message_snippet.html

```diff
@@ -1,6 +1,11 @@
 {% load static %}
 
-<div class="pb-5 message">
+<div class="pb-5 message"
+     x-data="{ overlay: false, state: 'initial' }"
+     hx-target="this"
+     hx-ext="alpine-morph"
+     hx-swap="morph"
+     @htmx:after-swap="state = 'flagged'">
```

Comment:
... the reason it was probably set on the list originally is that the whole message element gets replaced, and with it all the alpine state.

That's where this morph approach comes in, it means you can replace a whole element, but keep the alpine state.

Reply:
oh, interesting functionality by the morph plugin

Reply:
yeah, worth keeping in mind for other bits of the site that make lots of use of alpine state + htmx updates.